packages/ui/src/components/Table/Row.tsx

```diff
@@ -170,23 +171,20 @@ export const Row = ({
 
   const canClickRowToExpand = hasExpandable && !expandButton
 
-  useEffect(() => {
-    const { current } = checkboxRowRef
-
-    if (current) {
-      mapCheckbox.set(id, current)
-    }
-
-    return () => {
-      mapCheckbox.delete(id)
-    }
-  }, [mapCheckbox, id])
```

Comment:
Why changing this? Here you automatically register and unregister the checkbox ref into the list provider and update the list of it depending on rows. This is useful especially if you have dynamic rows which happens quite often (deleting an element, etc)